src/sentry/incidents/endpoints/organization_alert_rule_available_action_index.py

```diff
@@ -11,20 +11,43 @@
 from sentry.incidents.endpoints.serializers import action_target_type_to_string
 from sentry.incidents.logic import get_available_action_integrations_for_org
 from sentry.incidents.models import AlertRuleTriggerAction
+from sentry.models import OrganizationIntegration, PagerDutyService
 
 
 class OrganizationAlertRuleAvailableActionIndexEndpoint(OrganizationEndpoint):
-    def build_action_response(self, registered_type, integration=None):
-        return {
+    def fetch_services(self, organization, integration_id):
+        services = PagerDutyService.objects.filter(
+            organization_integration=OrganizationIntegration.objects.get(
+                organization=organization, integration=integration_id
+            )
+        )
+        formatted_services = [
+            {"value": service.id, "label": service.service_name} for service in services
+        ]
+        return formatted_services
+
+    def build_action_response(self, organization, registered_type, integration=None):
+        allowed_target_types = [
+            action_target_type_to_string[target_type]
+            for target_type in registered_type.supported_target_types
+        ]
+        input_type = ""
+        if "specific" in allowed_target_types:
+            input_type = "select" if registered_type.slug == "pagerduty" else "text"
```

Comment:
@ceorourke I wish there was better way to do this so we don't have as many code changes when we do this for OpsGenie.